pkg/network/tracer/connection/kprobe/config.go

```diff
@@ -57,7 +57,15 @@ func enabledProbes(c *config.Config, runtimeTracer bool) (map[probes.ProbeFuncNa
 		enableProbe(enabled, probes.InetCskAcceptReturn)
 		enableProbe(enabled, probes.InetCskListenStop)
 		enableProbe(enabled, probes.TCPSetState)
-		enableProbe(enabled, selectVersionBasedProbe(runtimeTracer, kv, probes.TCPRetransmit, probes.TCPRetransmitPre470, kv470))
+		// special case for tcp_retransmit_skb probe: on CO-RE,
+		// we want to load the version that makes use of
+		// the tcp_sock field, which is the same as the
+		// runtime compiled implementation
+		if coreTracer {
+			enableProbe(enabled, probes.TCPRetransmit)
+		} else {
+			enableProbe(enabled, selectVersionBasedProbe(runtimeTracer, kv, probes.TCPRetransmit, probes.TCPRetransmitPre470, kv470))
+		}
```

Comment:
Could this collapse to this? 
```suggestion
		enableProbe(enabled, selectVersionBasedProbe(runtimeTracer || coreTracer, kv, probes.TCPRetransmit, probes.TCPRetransmitPre470, kv470))
```